### PR TITLE
Add miniconda.sh to TRAVIS cache

### PR DESCRIPTION
**Pull request by notoraptor**
Trying to add miniconda.sh to TRAVIS cache.
The goal is to avoid re-downloading of this resource for every build.

Inspired from https://github.com/travis-ci/travis-ci/issues/1510#issuecomment-26274388
Need to be tested.

@nouiz

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -17,7 +18,7 @@ before_install:
 # Install miniconda to avoid compiling scipy
   - mkdir -p download
   - cd download
-  - wget -c https://repo.continuum.io/miniconda/Miniconda2-4.1.11-Linux-x86_64.sh -O miniconda.sh
+  - "[ -f $HOME/download/miniconda.sh ] || wget -c https://repo.continuum.io/miniconda/Miniconda2-4.1.11-Linux-x86_64.sh -O miniconda.sh"
```
Comment (nouiz) [inline .travis.yml]: You don't need to change this line. The wget -c parameter do that. It only redownload if needed.

**Comment by notoraptor** [comment]
Corrected.

**Comment by nouiz** [comment]
I'll merge now even if it don't seem to work. Maybe travis always reuse the master cache and the pr cache don't work or don't always work.

**Comment by nouiz** [comment]
I got this error in one build:

```
    sqlite:          3.13.0-0         

    tk:              8.5.18-0         

    wheel:           0.29.0-py27_0    

    zlib:            1.2.8-3          

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

The build has been terminated
```

I think the problem is the disc access the installation time take. Can you also add the installation to travis cache?

**Pull request merged**